faucet/src/faucet.rs

```diff
@@ -616,7 +626,31 @@ async fn break_up_records(state: FaucetState, mut wakeup: mpsc::Receiver<()>) {
             wakeup.next().await;
         }
 
-        // Start breaking up records until we have enough again.
+        if let Some(transactions) = get_more_records(&state).await {
+            // If we succeeded, wait until we are signalled again. Even though we may not have
+            // enough records just yet, we will when the transactions submitted by
+            // `get_more_records` finalize. Returning to the previous loop and checking if we have
+            // enough records might spuriously lead us to call `get_more_records` again, which would
+            // be an unnecessary waste of time.
+            tracing::info!(
+                "will have sufficient records after {} transactions, waiting for a change",
+                transactions.len()
+            );
+            wakeup.next().await;
+        }
+    }
+}
+
+/// Break records into smaller pieces to create at least `state.num_records` total.
+///
+/// If successful, returns a list of transaction receipts which will give at least
+/// `state.num_records` when they are finalized. If there were not enough large records to break up
+/// to obtain the desired number of records, returns [None].
+async fn get_more_records(state: &FaucetState) -> Option<Vec<TransactionReceipt<CapeLedger>>> {
```

Comment:
I think this would call this function `break_up_records` and the existing `break_up_records` could be called `maintain_faucet_records` or something like that.